### PR TITLE
Finer Options for Directions Generation

### PR DESCRIPTION
**Pull request by kevinkreiser**
Currently we have an option at request time called `narrative` which the docs described as:

> Boolean to allow you to disable narrative production. Locations, shape, length, and time are still returned. The narrative production is enabled by default. Set the value to `false` to disable the narrative.

After reading this, and indeed even after just reading the parameter name, I thought this meant that we would disable generating the instructions for the maneuvers but the maneuvers would still be present. The only clue that this is not the case is that the word maneuver is not mentioned in the items that are "still returned". So yeah this parameter disables all of both instructions and maneuver generation.

I would like to be able to skip instruction generation but still get maneuvers. The technical reason I want to do this is because std::locale is hard to deploy in real life. At least on linux it relies on posix locale files being available on LOCPATH. Long term we will probably want to move to using ICU for language things as its easier to support in a crossplatform way (and it has more features, though we dont need that many).

### CODE REVIEW

File: src/worker.cc [hunk 1]
```diff
@@ -471,9 +471,16 @@ void from_json(rapidjson::Document& doc, odin::DirectionsOptions& options) {
     options.set_language(*language);
   }
 
+  // deprecated
   auto narrative = rapidjson::get_optional<bool>(doc, "/narrative");
-  if (narrative) {
-    options.set_narrative(*narrative);
+  if (narrative && !*narrative) {
```
Comment (kevinkreiser) [inline src/worker.cc]: `"narrative":false` will still work to turn stuff off, we just dont reference it anymore from the docs

File: proto/directions_options.proto [hunk 2]
```diff
@@ -19,6 +19,12 @@ enum FilterAction {
   include = 1;
 }
 
+enum DirectionsType {
```
Comment (kevinkreiser) [inline proto/directions_options.proto]: the naming is kind of crappy here, any suggestions?

File: src/odin/directionsbuilder.cc [hunk 3]
```diff
@@ -78,20 +78,21 @@ TripDirections DirectionsBuilder::Build(const DirectionsOptions& directions_opti
...
-    // Create maneuvers
     ManeuversBuilder maneuversBuilder(directions_options, etp);
     maneuvers = maneuversBuilder.Build();
 
-    // Create the narrative
-    std::unique_ptr<NarrativeBuilder> narrative_builder =
-        NarrativeBuilderFactory::Create(directions_options, etp);
-    narrative_builder->Build(directions_options, etp, maneuvers);
+    // Create the instructions if desired
+    if (directions_options.directions_type() == DirectionsType::instructions) {
```
Comment (kevinkreiser) [inline src/odin/directionsbuilder.cc]: making the language generation part optional allows us to skip any usage of `std::locale` which means we can skip any exceptions that that will throw when the locales aren't present on the system. cc @frederoni

File: proto/directions_options.proto (hunk 2 above)
Comment (dgearhart) [reply proto/directions_options.proto]: @kevinkreiser i gave it serious thought and could think of anything better

**Pull request merged**